pkg/doc/verifiable/credential.go

```diff
@@ -223,16 +222,25 @@ type SchemaCache interface {
 	Get(k string) ([]byte, bool)
 }
 
+// cache defines a cache interface
+type cache interface {
+	Set(k, v []byte)
+
+	HasGet(dst, k []byte) ([]byte, bool)
+
+	Del(k []byte)
+}
+
 // ExpirableSchemaCache is an implementation of SchemaCache based fastcache.Cache with expirable elements.
 type ExpirableSchemaCache struct {
-	cache      *fastcache.Cache
+	cache      cache
 	expiration time.Duration
 }
 
 // NewExpirableSchemaCache creates new instance of ExpirableSchemaCache.
 func NewExpirableSchemaCache(size int, expiration time.Duration) *ExpirableSchemaCache {
 	return &ExpirableSchemaCache{
-		cache:      fastcache.New(size),
+		cache:      newCache(size),
```

Comment:
It almost seems like `NewExpirableSchemaCache` should have an argument of the cache interface type.

Reply:
I moved NewExpirableSchemaCache